server/src/main/scala/org/http4s/server/Fallthrough.scala

```diff
@@ -27,7 +27,8 @@ object Fallthrough {
   /** A [[Response]] specific [[Fallthrough]] which considers any response with a 404
     * status code as a fallthrough. */
   implicit def forResponse: Fallthrough[Response] = new Fallthrough[Response] {
-    def isFallthrough(r: Response): Boolean = r.status.code == 404
+    def isFallthrough(r: Response): Boolean =
+      r.status.code == 404 && r.attributes.contains(HttpService.notFoundServiceKey)
```

Comment:
Because a map lookup is quite costly compared to an integer comparison. I'm
not totally against it but I see it as the purview of an alternative
`Fallthrough`.

On Wed, Oct 7, 2015 at 10:32 AM, Rodolfo Hansen <notifications@github.com>
wrote:

> In server/src/main/scala/org/http4s/server/Fallthrough.scala
> <https://github.com/http4s/http4s/pull/425#discussion_r41397032>:
>
> > @@ -27,7 +27,8 @@ object Fallthrough {
> >    /** A [[Response]] specific [[Fallthrough]] which considers any response with a 404
> >      * status code as a fallthrough. */
> >    implicit def forResponse: Fallthrough[Response] = new Fallthrough[Response] {
> > -    def isFallthrough(r: Response): Boolean = r.status.code == 404
> > +    def isFallthrough(r: Response): Boolean =
> > +      r.status.code == 404 && r.attributes.contains(HttpService.notFoundServiceKey)
>
> Why not simply:
>
> def isFallthrough(r: Response): Boolean =
>       r.attributes.contains(HttpService.fallthroughServiceKey)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/http4s/http4s/pull/425/files#r41397032>.
>